packages/next-tinacms-cloudinary/src/handlers/create-media-handler.ts

```diff
@@ -0,0 +1,121 @@
+import { v2 as cloudinary } from 'cloudinary'
+import { Media, MediaListOptions } from 'tinacms'
+import path from 'path'
+import { NextApiRequest, NextApiResponse } from 'next'
+import multer from 'multer'
+import { promisify } from 'util'
+
+export interface CloudinaryConfig {
+  cloud_name: string
+  api_key: string
+  api_secret: string
+}
+
+export const mediaHandlerConfig = {
+  api: {
+    bodyParser: false,
+  },
+}
+
+export const createMediaHandler = (config: CloudinaryConfig) => {
+  cloudinary.config(config)
+
+  return async (req: NextApiRequest, res: NextApiResponse) => {
+    switch (req.method) {
+      case 'GET':
+        return listMedia(req, res)
+      case 'POST':
+        return uploadMedia(req, res)
```

Comment:
I think this is all we need to totally remove the `/media/[public_id].ts` file:
```suggestion
        return uploadMedia(req, res)
    case 'DELETE':
      const { media } = req.query
      const [_media, public_id] = media
      cloudinary.uploader.destroy(public_id as string, {}, (err) => {
        if (err) res.status(500)
        res.json({
          err,
          public_id,
        })
      })
```

Reply:
Latest changes should update [...media].ts as a catch all (GET, POST, DELETE)

Also, cleans up the media store class and removes the useless constructor.